src/openforms/registrations/contrib/json/config.py

```diff
@@ -0,0 +1,51 @@
+from typing import Required, TypedDict
+
+from django.utils.translation import gettext_lazy as _
+
+from rest_framework import serializers
+from zgw_consumers.constants import APITypes
+from zgw_consumers.models import Service
+
+from openforms.api.fields import PrimaryKeyRelatedAsChoicesField
+from openforms.formio.api.fields import FormioVariableKeyField
+from openforms.utils.mixins import JsonSchemaSerializerMixin
+
+
+class JSONOptionsSerializer(JsonSchemaSerializerMixin, serializers.Serializer):
+    service = PrimaryKeyRelatedAsChoicesField(
+        queryset=Service.objects.filter(api_type=APITypes.orc),
+        label=_("Service"),
+        help_text=_("Which service to use."),
+        required=True,
+    )
+    relative_api_endpoint = serializers.CharField(
+        max_length=255,
+        label=_("Relative API endpoint"),
+        help_text=_(
+            "The API endpoint to send the data to (relative to the service API root)."
+        ),
+        allow_blank=True,
+        required=False,
+    )
+    form_variables = serializers.ListField(
+        child=FormioVariableKeyField(max_length=50),
+        label=_("Form variable key list"),
+        help_text=_(
+            "A list of form variables (can also include static variables) to use."
+        ),
+        required=True,
+        min_length=1,
+    )
+
+
+class JSONOptions(TypedDict):
+    """
+    JSON registration plugin options
+
+    This describes the shape of :attr:`JSONOptionsSerializer.validated_data`, after
+    the input data has been cleaned/validated.
+    """
+
+    service: Required[Service]
+    relative_api_endpoint: str
+    form_variables: Required[list[str]]
```

Comment:
```suggestion
    service: Service
    relative_api_endpoint: str
    form_variables: list[str]
```
`Required` is the default, unless you specify `total=False` on the `TypedDict`.

If you specify `default=""` on the relative API endpoint field, then you also have the guarantee that the key will always be present, and that gives the most ergonomic type definition (nice and strict).